### PR TITLE
Added TxDependency in block header (metadata for parallel execution)

### PR DESCRIPTION
**Pull request by pratikspatil024**

### CODE REVIEW

File: core/types/block_test.go [hunk 1]
```diff
@@ -76,6 +76,266 @@ func TestBlockEncoding(t *testing.T) {
...
+	}
+
+	block.header = header
+
+	_, err := rlp.EncodeToBytes(&block)
+	if err != nil {
+		t.Fatal("encode error: ", err)
+	}
+}
+
```
Comment (cffls) [inline core/types/block_test.go:338]: Could you add a test case that verifies tx dependency is nil after decoding a regular tx?

Comment (pratikspatil024) [reply core/types/block_test.go:338]: Will look into this, thanks!

Comment (pratikspatil024) [reply core/types/block_test.go:405]: Is [this](https://github.com/maticnetwork/erigon/pull/18/commits/1216b1304aa621b4af6eb2a1bdb4a7c408a11524) okay?

Comment (cffls) [reply core/types/block_test.go:405]: Looks good, thank you for adding it!

**Review by cffls — APPROVED** [review]
